packages/sn-controls-react/src/fieldcontrols/Number.tsx

```diff
@@ -3,111 +3,91 @@
-            type="number"
-            label={this.props.settings.DisplayName}
-            value={this.state.value}
-            required={this.props.settings.Compulsory}
-            disabled={this.props.settings.ReadOnly}
-            InputProps={{
-              startAdornment: this.defineCurrency(),
-              endAdornment: this.props.settings.ShowAsPercentage ? (
-                <InputAdornment position="end">%</InputAdornment>
-              ) : null,
-            }}
-            inputProps={{
-              step: this.defineStepValue(),
-              max: this.props.settings.MaxValue,
-              min: this.props.settings.MinValue,
-            }}
-            id={this.props.settings.Name}
-            fullWidth={true}
-            onChange={this.handleChange}
-            helperText={this.props.settings.Description}
-          />
-        )
-      case 'browse':
-      default:
-        return this.state.value ? (
-          <div>
-            <Typography variant="caption" gutterBottom={true}>
-              {this.props.settings.DisplayName}
-            </Typography>
-            <Typography variant="body1" gutterBottom={true}>
-              {isCurrencyFieldSetting(this.props.settings)
-                ? this.props.settings.Format
-                  ? this.props.settings.Format
-                  : '$'
-                : null}
-              {this.state.value}
-              {this.props.settings.ShowAsPercentage ? '%' : null}
-            </Typography>
-          </div>
-        ) : null
-    }
+  switch (props.actionName) {
+    case 'edit':
+    case 'new':
+      return (
+        <TextField
+          name={props.settings.Name}
+          type="number"
+          label={props.settings.DisplayName}
+          value={value}
+          required={props.settings.Compulsory}
+          disabled={props.settings.ReadOnly}
+          placeholder={props.settings.DisplayName}
+          InputProps={{
+            startAdornment: defineCurrency(),
+            endAdornment: props.settings.ShowAsPercentage ? <InputAdornment position="end">%</InputAdornment> : null,
+          }}
+          inputProps={{
+            step: defineStepValue(),
+            max: props.settings.MaxValue,
+            min: props.settings.MinValue,
+          }}
+          id={props.settings.Name}
+          fullWidth={true}
+          onChange={handleChange}
+          helperText={props.settings.Description}
+        />
+      )
+    case 'browse':
+    default:
+      return props.fieldValue ? (
+        <div>
+          <Typography variant="caption" gutterBottom={true}>
+            {props.settings.DisplayName}
+          </Typography>
+          <Typography variant="body1" gutterBottom={true}>
+            {isCurrencyFieldSetting(props.settings) ? (props.settings.Format ? props.settings.Format : '$') : null}
+            {props.fieldValue}
+            {props.settings.ShowAsPercentage ? '%' : null}
+          </Typography>
+        </div>
+      ) : null
```

Comment:
Maybe an example in storybook with double value and double value as Step would be nice.